relayer/provider/provider.go

```diff
@@ -29,16 +29,22 @@ type RelayerTxResponse struct {
 	TxHash string
 	Code   uint32
 	Data   string
-	Events map[string]string
+	Events []*RelayerEvent
+}
+
+type RelayerEvent struct {
+	EventType      string
+	AttributeKey   string
+	AttributeValue string
 }
 
 func (r RelayerTxResponse) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 	enc.AddInt64("height", r.Height)
 	enc.AddString("tx_hash", r.TxHash)
 	enc.AddUint32("code", r.Code)
 	enc.AddString("data", r.Data)
-	for k, v := range r.Events {
-		enc.AddString("event:"+k, v)
+	for _, event := range r.Events {
+		enc.AddString("event:"+event.EventType, event.AttributeKey+"="+event.AttributeValue)
```

Comment:
First I was going to suggest that you could change the field key here to `event.EventType + ":" + event.AttributeKey`, but you have said that a RelayerTxResponse may have multiple events with the same type and key.

For structured logging, to deal with expected-to-overlap keys, it looks like we should use an array to match the underlying type. Something like `enc.AddArray("events", loggableEvents(r.Events))` with:

```go
// MarshalObject satisfies the zapcore.ObjectMarshaler interface.
func (e RelayerEvent) MarshalObject(enc zapcore.ObjectEncoder) error {
  enc.AddString("event_type", event.EventType)
  enc.AddString("key", event.AttributeKey)
  enc.AddString("value", event.AttributeValue)
  return nil
}

// loggableEvents is an unexported wrapper type for a slice of RelayerEvent,
// to satisfy the zapcore.ArrayMarshaler interface.
type loggableEvents []RelayerEvent

func (es loggableEvents) MarshalLogArray(enc zapcore.ArrayEncoder) error {
  for _, e := range es {
    enc.AppendObject(e)
  }
  return nil
}
```